### PR TITLE
SavedSearch - Add name and label columns

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Adds 2 database columns: `name` and `label` for the SavedSearch entity.
This is needed by the new Search Kit extension.

Before
--------------
`SavedSearch` entity lacks unique name or label.
Pre/post hooks not called for SavedSearch entity.

After
----------------------------------------
Label column added for identifying saved search on administrative screens.
Unique name column added for embedding savedsearch displays by name on pages, dashlets, etc.
Pre/post hooks called.

Technical Details
----------------------------------------
Unique name generated from label if not supplied.
Switching from ad-hoc create function to the standard `dao::writeRecord` ensures that hooks are properly called.

### CODE REVIEW

**Comment by colemanw** [comment]
@seamuslee001 @eileenmcnaughton @totten this is going to go stale real fast.

File: xml/schema/Contact/SavedSearch.xml [hunk 1]
```diff
@@ -18,6 +18,39 @@
...
+  </field>
+  <index>
+    <name>UI_name</name>
+    <fieldName>name</fieldName>
+    <unique>true</unique>
+    <add>5.32</add>
+  </index>
+
+  <field>
+    <name>label</name>
```
Comment (demeritcowboy) [inline xml/schema/Contact/SavedSearch.xml:42]: Does this need "localizable"?

File: CRM/Upgrade/Incremental/php/FiveThirtyTwo.php [hunk 2]
```diff
@@ -46,27 +47,15 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
-  //    // The above is an exception because 'Upgrade DB to %1: SQL' is generic & reusable.
-  //  }
-
-  // public static function taskFoo(CRM_Queue_TaskContext $ctx, ...) {
-  //   return TRUE;
-  // }
+  public function upgrade_5_32_alpha1($rev) {
+    $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('Add column civicrm_saved_search.name', 'addColumn', 'civicrm_saved_search', 'name', "varchar(255)   DEFAULT NULL COMMENT 'Unique name of saved search'");
+    $this->addTask('Add column civicrm_saved_search.label', 'addColumn', 'civicrm_saved_search', 'label', "varchar(255)   DEFAULT NULL COMMENT 'Administrative label for search'");
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/php/FiveThirtyTwo.php:58]: If it needs localizable for label, then also the localizable param here needs to be TRUE.

File: xml/schema/Contact/SavedSearch.xml (hunk 1 above)
Comment (colemanw) [reply xml/schema/Contact/SavedSearch.xml:42]: No I don't think so. This is intended for back-end administration screens.

**Comment by colemanw** [comment]
Test fails were being caused by the sloppy function `CRM_Contact_BAO_Group::createSmartGroup()`. I cleaned it up to fix the failures, but it's currently only ever used by tests, so we could delete it and refactor the tests to use something else, or move it into the test class and out of the BAO.

**Comment by colemanw** [comment]
@eileenmcnaughton I forgot to add the unique index to the upgrader. Looking at the function `CRM_Upgrade_Incremental_Base::addIndex()` it doesn't appear capable of creating a UNIQUE index.
But I see there's also a function `CRM_Core_BAO_SchemaHandler::createMissingIndices()` which looks like it would automatically add the index based on the schema... but I don't see it being called from the upgrader? Is it ever called routinely, or only manually?

**Comment by eileenmcnaughton** [comment]
@colemanw only manually

**Pull request merged**